Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -2282,6 +2282,14 @@ LayoutUnit RenderBox::perpendicularContainingBlockLogicalHeight() const
     return cb->adjustContentBoxLogicalHeightForBoxSizing(LayoutUnit(logicalHeightLength.value()));
 }
 
+bool RenderBox::isAdjacentToBlockContainerBeforeEdge() const
+{
+    if (needsLayout())
+        return false;
+    auto* container = containingBlock();
+    return container->isBlockContainer() && container->logicalMarginBoxTopForChild(*this) == container->borderAndPaddingBefore();
+}
```

Comment:
I am not sure if this should be a standalone function. I think it only makes sense in the context of margin trimming. I'd wrap it inside  hasTrimmedBlockStartMarginInBlockContainer

Reply:
Moved